### PR TITLE
Improve handling of mismatched Swift versions

### PR DESCRIPTION
**Pull request by DanToml**
\c @benasher44

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,14 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Ben Asher](https://github.com/benasher44)
   [#5959](https://github.com/CocoaPods/CocoaPods/pull/5959)
 
+* Error with helpful message when integrating a pod into targets that have mismatched Swift versions.  
+  [Ben Asher](https://github.com/benasher44)
```
Comment (benasher44) [inline CHANGELOG.md]: <3

File: lib/cocoapods/installer/analyzer.rb [hunk 2]
```diff
@@ -406,6 +407,39 @@ def generate_target(target_definition, pod_targets)
...
+            (targets_by_spec[spec] ||= []) << target
+          end
+        end
+
+        error_message_for_target = lambda do |target|
+          "#{target.name} (Swift #{target.swift_version})"
+        end
+
+        error_messages = targets_by_spec.map do |spec, targets|
+          swift_targets = targets.reject { |target| target.swift_version.blank? }
```
Comment (benasher44) [inline lib/cocoapods/installer/analyzer.rb]: swift_version can be nil in some scenarios, and I think we want to reject that here as well. I made these notes in an early commit in my previous PR:
```
# when the swift version is unset at the project level, but set in one target, swift_version is nil
# when the swift version is set at the project level, but unset in one target, swift_version is empty
```

Comment (benasher44) [inline lib/cocoapods/installer/analyzer.rb]: Maybe `TargetDefinition` could normalize this?

Comment (DanToml) [inline lib/cocoapods/installer/analyzer.rb]: Blank?  Checks for nil and empty.  I think there is actually a semantic difference between them in Xcode so I'm not so sure that we should normalise?

Comment (benasher44) [inline lib/cocoapods/installer/analyzer.rb]: Ahhh. I had a suspicion that was the case. I couldn't find the docs for it, but now I se active support adds this. 👍

**Pull request merged**